### DISCUSSION TITLE
Promise.all() with getAttributes()

### DISCUSSION THREAD

**Discussion by iMaupin** [wix/Detox#3418] [Q&A]
In more recent versions of detox, it exits tests if it detects concurrent interactions. Usually, this is a welcome debugging tool to ensure there aren't overlooked await keywords. 

It is nice, however, to make getAttributes calls concurrently when fetching multiple attributes. Is there a workaround to make specifically enable Promise.all() statements to not throw this error?

**Comment by noomorph** [comment]
@d4vidi, 😬  WDYT?

**Comment by d4vidi** [comment]
It's a nice optimization, but pretty risky, as there's no easy way to guarantee `getAttributes()` is not called concurrently with other actions, such as a `tap()` that pushed a screen, for example.
